packages/eas-cli/src/uploads.ts

```diff
@@ -27,10 +32,70 @@ export async function uploadFileAtPathToS3Async(
   return { url, bucketKey: presignedPost.fields.key };
 }
 
-export async function uploadWithPresignedPostAsync(
+export async function uploadWithPresignedPostWithRetryAsync(
+  file: string,
+  presignedPost: PresignedPost
+): Promise<Response> {
+  return await promiseRetry(
+    async retry => {
+      // retry fetch errors (usually connection or DNS errors)
+      let response: Response;
+      try {
+        response = await uploadWithPresignedPostAsync(file, presignedPost);
+      } catch (e: any) {
+        return retry(e);
+      }
+
+      // retry 408, 429, 5xx as suggested by google
+      if (
+        response.status === 408 ||
+        response.status === 429 ||
+        Math.floor(response.status / 100) === 5 // 5xx errors
+      ) {
+        return retry(new Error(`Presigned upload responded with a ${response.status} status`));
+      }
+
+      // don't retry other errors
+      if (!response.ok) {
+        throw new Error(`Presigned upload responded with a ${response.status} status`);
+      }
+
+      return response;
+    },
+    // retry parameters match google suggested defaults: https://cloud.google.com/storage/docs/retry-strategy#node.js
+    {
+      retries: 3,
+      factor: 2,
+    }
+  );
+}
+
+async function uploadWithPresignedPostAsync(
```

Comment:
Might be overoptimizing for code deduplication but one clean way to reduce duplicate code between this method and uploadWithPresignedPostWithProgressAsync would be to change this method to `createPresignedPostRequest: Request`, which returns an object that can be passed to fetch().

uploadWithPresignedPostWithProgressAsync can get the FormData object from the Request to add listeners to it before passing the Request to fetch().